src/service/viewer-impl.js

```diff
@@ -835,6 +835,34 @@ export class Viewer {
   }
 
   /**
+   *
+   * @return {!Promise}
+   */
+  replaceShareTrackingFragment(incomingFragment, outgoingFragment) {
```

Comment:
It will update the URL fragment.

A separation will be something like the following:
```
const originFragment = viewer.getFragment();
const newFragment = getNewFragment(originFragment);
viewer.updateFragment(newFragment);
```